In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '현대고등학교 건너편']
df

자치구        대여소명  대여소번호     년   월  시간     lambda         mu     c  \
671328  강남구  현대고등학교 건너편   2301  2020   6   0   6.066667   5.533333  13.0   
671329  강남구  현대고등학교 건너편   2301  2020   6   1   3.800000   4.266667  13.0   
671330  강남구  현대고등학교 건너편   2301  2020   6   2   2.566667   2.833333  13.0   
671331  강남구  현대고등학교 건너편   2301  2020   6   3   1.900000   1.466667  13.0   
671332  강남구  현대고등학교 건너편   2301  2020   6   4   1.766667   1.200000  13.0   
671333  강남구  현대고등학교 건너편   2301  2020   6   5   0.966667   0.933333  13.0   
671334  강남구  현대고등학교 건너편   2301  2020   6   6   1.166667   1.066667  13.0   
671335  강남구  현대고등학교 건너편   2301  2020   6   7   1.366667   1.400000  13.0   
671336  강남구  현대고등학교 건너편   2301  2020   6   8   3.333333   2.466667  13.0   
671337  강남구  현대고등학교 건너편   2301  2020   6   9   5.366667   2.966667  13.0   
671338  강남구  현대고등학교 건너편   2301  2020   6  10   3.533333   2.233333  13.0   
671339  강남구  현대고등학교 건너편   2301  2020   6  11   3.700000   3.166667  13.0   
671340  강남구  현대고등학교 건너편   2301  2020   6  12   4.966667   4.266667  13.0   
671341  강남구  현대고등학교 건너편   2301  2020   6  13   4.300000   2.966667  13.0   
671342  강남구  현대고등학교 건너편   2301  2020   6  14   4.600000   4.633333  13.0   
671343  강남구  현대고등학교 건너편   2301  2020   6  15   6.433333   7.133333  13.0   
671344  강남구  현대고등학교 건너편   2301  2020   6  16   7.800000   9.333333  13.0   
671345  강남구  현대고등학교 건너편   2301  2020   6  17  10.333333  10.466667  13.0   
671346  강남구  현대고등학교 건너편   2301  2020   6  18  16.366667  17.600000  13.0   
671347  강남구  현대고등학교 건너편   2301  2020   6  19  14.666667  16.000000  13.0   
671348  강남구  현대고등학교 건너편   2301  2020   6  20  16.133333  14.466667  13.0   
671349  강남구  현대고등학교 건너편   2301  2020   6  21  12.933333  13.600000  13.0   
671350  강남구  현대고등학교 건너편   2301  2020   6  22  12.133333  10.533333  13.0   
671351  강남구  현대고등학교 건너편   2301  2020   6  23   9.666667   7.533333  13.0   
671352  강남구  현대고등학교 건너편   2301  2020   7   0   4.800000   4.766667  13.0   
671353  강남구  현대고등학교 건너편   2301  2020   7   1   3.466667   3.133333  13.0   
671354  강남구  현대고등학교 건너편   2301  2020   7   2   2.666667   3.000000  13.0   
671355  강남구  현대고등학교 건너편   2301  2020   7   3   1.666667   1.700000  13.0   
671356  강남구  현대고등학교 건너편   2301  2020   7   4   1.600000   0.866667  13.0   
671357  강남구  현대고등학교 건너편   2301  2020   7   5   0.533333   1.133333  13.0   
671358  강남구  현대고등학교 건너편   2301  2020   7   6   0.466667   0.666667  13.0   
671359  강남구  현대고등학교 건너편   2301  2020   7   7   1.133333   0.833333  13.0   
671360  강남구  현대고등학교 건너편   2301  2020   7   8   1.966667   1.266667  13.0   
671361  강남구  현대고등학교 건너편   2301  2020   7   9   2.866667   1.700000  13.0   
671362  강남구  현대고등학교 건너편   2301  2020   7  10   3.100000   1.633333  13.0   
671363  강남구  현대고등학교 건너편   2301  2020   7  11   2.366667   2.166667  13.0   
671364  강남구  현대고등학교 건너편   2301  2020   7  12   4.233333   3.200000  13.0   
671365  강남구  현대고등학교 건너편   2301  2020   7  13   4.266667   4.433333  13.0   
671366  강남구  현대고등학교 건너편   2301  2020   7  14   4.833333   4.766667  13.0   
671367  강남구  현대고등학교 건너편   2301  2020   7  15   6.333333   6.833333  13.0   
671368  강남구  현대고등학교 건너편   2301  2020   7  16   8.233333   8.833333  13.0   
671369  강남구  현대고등학교 건너편   2301  2020   7  17   9.433333  11.266667  13.0   
671370  강남구  현대고등학교 건너편   2301  2020   7  18  12.000000  12.733333  13.0   
671371  강남구  현대고등학교 건너편   2301  2020   7  19  12.566667  12.600000  13.0   
671372  강남구  현대고등학교 건너편   2301  2020   7  20  13.566667  13.166667  13.0   
671373  강남구  현대고등학교 건너편   2301  2020   7  21  12.266667  13.166667  13.0   
671374  강남구  현대고등학교 건너편   2301  2020   7  22  11.833333  10.000000  13.0   
671375  강남구  현대고등학교 건너편   2301  2020   7  23   7.800000   7.133333  13.0   
671376  강남구  현대고등학교 건너편   2301  2020   8   0   2.366667   1.633333  13.0   
671377  강남구  현대고등학교 건너편   2301  2020   8   1   1.866667   1.700000  13.0   
671378  강남구  현대고등학교 건너편   2301  2020   8   2   1.666667   1.233333  13.0   
671379  강남구  현대고등학교 건너편   2301  2020   8   3   1.066667 

In [4]:
df = df.drop(['대여소번호', 'lambda', 'mu', 'c', 'rho', 'p_c', 'p_1+'], axis = 1)
df

자치구        대여소명     년   월  시간           p_0           p_1        L_s  \
671328  강남구  현대고등학교 건너편  2020   6   0  3.669750e-02  4.023461e-02   7.955313   
671329  강남구  현대고등학교 건너편  2020   6   1  1.363053e-01  1.213969e-01   4.695780   
671330  강남구  현대고등학교 건너편  2020   6   2  1.255930e-01  1.137725e-01   4.943041   
671331  강남구  현대고등학교 건너편  2020   6   3  8.096958e-03  1.048924e-02   9.999056   
671332  강남구  현대고등학교 건너편  2020   6   4  2.110844e-03  3.107632e-03  10.944933   
671333  강남구  현대고등학교 건너편  2020   6   5  5.629577e-02  5.830633e-02   7.067942   
671334  강남구  현대고등학교 건너편  2020   6   6  3.740527e-02  4.091202e-02   7.919187   
671335  강남구  현대고등학교 건너편  2020   6   7  8.314769e-02  8.116798e-02   6.109159   
671336  강남구  현대고등학교 건너편  2020   6   8  5.265578e-03  7.115646e-03  10.363659   
671337  강남구  현대고등학교 건너편  2020   6   9  2.013485e-04  3.642372e-04  11.767373   
671338  강남구  현대고등학교 건너편  2020   6  10  9.472423e-04  1.498622e-03  11.304834   
671339  강남구  현대고등학교 건너편  2020   6  11  2.148568e-02  2.510432e-02   8.848497   
671340  강남구  현대고등학교 건너편  2020   6  12  2.220605e-02  2.584923e-02   8.799678   
671341  강남구  현대고등학교 건너편  2020   6  13  2.501987e-03  3.626476e-03  10.852937   
671342  강남구  현대고등학교 건너편  2020   6  14  7.482904e-02  7.429070e-02   6.382691   
671343  강남구  현대고등학교 건너편  2020   6  15  1.283612e-01  1.157650e-01   4.877612   
671344  강남구  현대고등학교 건너편  2020   6  16  1.787775e-01  1.494069e-01   3.852003   
671345  강남구  현대고등학교 건너편  2020   6  17  7.753251e-02  7.654483e-02   6.291776   
671346  강남구  현대고등학교 건너편  2020   6  18  1.097732e-01  1.020807e-01   5.339372   
671347  강남구  현대고등학교 건너편  2020   6  19  1.183332e-01  1.084721e-01   5.120015   
671348  강남구  현대고등학교 건너편  2020   6  20  3.198125e-02  3.566572e-02   8.206361   
671349  강남구  현대고등학교 건너편  2020   6  21  9.702402e-02  9.226794e-02   5.689941   
671350  강남구  현대고등학교 건너편  2020   6  22  2.433874e-02  2.803576e-02   8.659887   
671351  강남구  현대고등학교 건너편  2020   6  23  8.901409e-03  1.142216e-02   9.908813   
671352  강남구  현대고등학교 건너편  2020   7   0  6.823837e-02  6.871556e-02   6.613223   
671353  강남구  현대고등학교 건너편  2020   7   1  3.411997e-02  3.774975e-02   8.090188   
671354  강남구  현대고등학교 건너편  2020   7   2  1.375562e-01  1.222722e-01   4.667920   
671355  강남구  현대고등학교 건너편  2020   7   3  8.098232e-02  7.939444e-02   6.178621   
671356  강남구  현대고등학교 건너편  2020   7   4  1.584091e-04  2.924476e-04  11.820803   
671357  강남구  현대고등학교 건너편  2020   7   5  5.294256e-01  2.491415e-01   0.888523   
671358  강남구  현대고등학교 건너편  2020   7   6  3.020486e-01  2.114340e-01   2.237734   
671359  강남구  현대고등학교 건너편  2020   7   7  4.927905e-03  6.701951e-03  10.413863   
671360  강남구  현대고등학교 건너편  2020   7   8  1.170568e-03  1.817462e-03  11.220131   
671361  강남구  현대고등학교 건너편  2020   7   9  4.568683e-04  7.704054e-04  11.552177   
671362  강남구  현대고등학교 건너편  2020   7  10  1.141026e-04  2.165621e-04  11.888143   
671363  강남구  현대고등학교 건너편  2020   7  11  3.779770e-02  4.128672e-02   7.899318   
671364  강남구  현대고등학교 건너편  2020   7  12  6.551775e-03  8.667453e-03  10.187277   
671365  강남구  현대고등학교 건너편  2020   7  13  9.054699e-02  8.714297e-02   5.880300   
671366  강남구  현대고등학교 건너편  2020   7  14  6.516039e-02  6.607173e-02   6.725555   
671367  강남구  현대고등학교 건너편  2020   7  15  1.117351e-01  1.035593e-01   5.288024   
671368  강남구  현대고등학교 건너편  2020   7  16  1.084227e-01  1.010581e-01   5.375102   
671369  강남구  현대고등학교 건너편  2020   7  17  1.774909e-01  1.486093e-01   3.874779   
671370  강남구  현대고등학교 건너편  2020   7  18  1.020876e-01  9.620824e-02   5.547061   
671371  강남구  현대고등학교 건너편  2020   7  19  7.266497e-02  7.247274e-02   6.456955   
671372  강남구  현대고등학교 건너편  2020   7  20  5.837591e-02  6.014935e-02   6.984896   
671373  강남구  현대고등학교 건너편  2020   7  21  1.086917e-01  1.012621e-01   5.367959   
671374  강남구  현대고등학교 건너편  2020   7  22  1.918515e-02  2.270243e-02   9.010503   
671375  강남구  현대고등학교 건너편  2020   7  23  3.748444e-02  4.098766e-02   7.915170   
671376  강남구  현대고등학교 건너편  2020   8   0  2.510594e-03  3.637800e-03  10.851012   
671377  

In [5]:
data = pd.read_csv('./현대고등학교 건너편 균형, 불균형.csv', encoding = 'cp949')
data

자치구        대여소명     년   월  시간        재고량      3사분위수   1사분위수   균형
0    강남구  현대고등학교 건너편  2020   6   0  17.100000  20.008333  11.025   균형
1    강남구  현대고등학교 건너편  2020   6   1  16.733333  20.008333  11.025   균형
2    강남구  현대고등학교 건너편  2020   6   2  16.800000  20.008333  11.025   균형
3    강남구  현대고등학교 건너편  2020   6   3  17.266667  20.008333  11.025   균형
4    강남구  현대고등학교 건너편  2020   6   4  17.833333  20.008333  11.025   균형
5    강남구  현대고등학교 건너편  2020   6   5  18.266667  20.008333  11.025   균형
6    강남구  현대고등학교 건너편  2020   6   6  18.700000  20.008333  11.025   균형
7    강남구  현대고등학교 건너편  2020   6   7  18.766667  20.008333  11.025   균형
8    강남구  현대고등학교 건너편  2020   6   8  10.466667  20.008333  11.025  불균형
9    강남구  현대고등학교 건너편  2020   6   9  12.866667  20.008333  11.025   균형
10   강남구  현대고등학교 건너편  2020   6  10  14.200000  20.008333  11.025   균형
11   강남구  현대고등학교 건너편  2020   6  11  14.766667  20.008333  11.025   균형
12   강남구  현대고등학교 건너편  2020   6  12  15.466667  20.008333  11.025   균형
13   강남구  현대고등학교 건너편  2020   6  13  17.033333  20.008333  11.025   균형
14   강남구  현대고등학교 건너편  2020   6  14  17.266667  20.008333  11.025   균형
15   강남구  현대고등학교 건너편  2020   6  15  16.566667  20.008333  11.025   균형
16   강남구  현대고등학교 건너편  2020   6  16  15.066667  20.008333  11.025   균형
17   강남구  현대고등학교 건너편  2020   6  17  14.933333  20.008333  11.025   균형
18   강남구  현대고등학교 건너편  2020   6  18  13.700000  20.008333  11.025   균형
19   강남구  현대고등학교 건너편  2020   6  19  12.400000  20.008333  11.025   균형
20   강남구  현대고등학교 건너편  2020   6  20  14.066667  20.008333  11.025   균형
21   강남구  현대고등학교 건너편  2020   6  21  13.400000  20.008333  11.025   균형
22   강남구  현대고등학교 건너편  2020   6  22  15.000000  20.008333  11.025   균형
23   강남구  현대고등학교 건너편  2020   6  23  17.133333  20.008333  11.025   균형
24   강남구  현대고등학교 건너편  2020   7   0  10.433333  20.008333  11.025  불균형
25   강남구  현대고등학교 건너편  2020   7   1  10.966667  20.008333  11.025  불균형
26   강남구  현대고등학교 건너편  2020   7   2  11.133333  20.008333  11.025   균형
27   강남구  현대고등학교 건너편  2020   7   3  11.566667  20.008333  11.025   균형
28   강남구  현대고등학교 건너편  2020   7   4  12.400000  20.008333  11.025   균형
29   강남구  현대고등학교 건너편  2020   7   5  12.200000  20.008333  11.025   균형
30   강남구  현대고등학교 건너편  2020   7   6  12.300000  20.008333  11.025   균형
31   강남구  현대고등학교 건너편  2020   7   7  12.666667  20.008333  11.025   균형
32   강남구  현대고등학교 건너편  2020   7   8   7.366667  20.008333  11.025  불균형
33   강남구  현대고등학교 건너편  2020   7   9   8.600000  20.008333  11.025  불균형
34   강남구  현대고등학교 건너편  2020   7  10  10.066667  20.008333  11.025  불균형
35   강남구  현대고등학교 건너편  2020   7  11  10.300000  20.008333  11.025  불균형
36   강남구  현대고등학교 건너편  2020   7  12  11.333333  20.008333  11.025   균형
37   강남구  현대고등학교 건너편  2020   7  13  11.266667  20.008333  11.025   균형
38   강남구  현대고등학교 건너편  2020   7  14  11.333333  20.008333  11.025   균형
39   강남구  현대고등학교 건너편  2020   7  15  11.100000  20.008333  11.025   균형
40   강남구  현대고등학교 건너편  2020   7  16  10.500000  20.008333  11.025  불균형
41   강남구  현대고등학교 건너편  2020   7  17   8.700000  20.008333  11.025  불균형
42   강남구  현대고등학교 건너편  2020   7  18   8.033333  20.008333  11.025  불균형
43   강남구  현대고등학교 건너편  2020   7  19   8.000000  20.008333  11.025  불균형
44   강남구  현대고등학교 건너편  2020   7  20   8.400000  20.008333  11.025  불균형
45   강남구  현대고등학교 건너편  2020   7  21   7.633333  20.008333  11.025  불균형
46   강남구  현대고등학교 건너편  2020   7  22   9.533333  20.008333  11.025  불균형
47   강남구  현대고등학교 건너편  2020   7  23  10.200000  20.008333  11.025  불균형
48   강남구  현대고등학교 건너편  2020   8   0  13.033333  20.008333  11.025   균형
49   강남구  현대고등학교 건너편  2020   8   1  13.433333  20.008333  11.025   균형
50   강남구  현대고등학교 건너편  2020   8   2  13.966667  20.008333  11.025   균형
51   강남구  현대고등학교 건너편  2020   8   3  14.333333  20.008333  11.025   균형
52   강남구  현대고등학교 건너편  2020   8   4  14.566667  20.008333  11.025   균형
53   강남구  현대고등학교 건너편  2020   8   5  14.600000  20.008333  11.025   균형
54   강남구  현대고등학교 건너편  2020   8   6  14.700000  20.008333  11.025   균형
55   강남구  현대고등학교 건너편  2020   8   7  14.866667  20.008333  11.025   균형
56   강남구  현대고등학

In [6]:
data = data.drop(['3사분위수', '1사분위수'], axis = 1)
data

자치구        대여소명     년   월  시간        재고량   균형
0    강남구  현대고등학교 건너편  2020   6   0  17.100000   균형
1    강남구  현대고등학교 건너편  2020   6   1  16.733333   균형
2    강남구  현대고등학교 건너편  2020   6   2  16.800000   균형
3    강남구  현대고등학교 건너편  2020   6   3  17.266667   균형
4    강남구  현대고등학교 건너편  2020   6   4  17.833333   균형
5    강남구  현대고등학교 건너편  2020   6   5  18.266667   균형
6    강남구  현대고등학교 건너편  2020   6   6  18.700000   균형
7    강남구  현대고등학교 건너편  2020   6   7  18.766667   균형
8    강남구  현대고등학교 건너편  2020   6   8  10.466667  불균형
9    강남구  현대고등학교 건너편  2020   6   9  12.866667   균형
10   강남구  현대고등학교 건너편  2020   6  10  14.200000   균형
11   강남구  현대고등학교 건너편  2020   6  11  14.766667   균형
12   강남구  현대고등학교 건너편  2020   6  12  15.466667   균형
13   강남구  현대고등학교 건너편  2020   6  13  17.033333   균형
14   강남구  현대고등학교 건너편  2020   6  14  17.266667   균형
15   강남구  현대고등학교 건너편  2020   6  15  16.566667   균형
16   강남구  현대고등학교 건너편  2020   6  16  15.066667   균형
17   강남구  현대고등학교 건너편  2020   6  17  14.933333   균형
18   강남구  현대고등학교 건너편  2020   6  18  13.700000   균형
19   강남구  현대고등학교 건너편  2020   6  19  12.400000   균형
20   강남구  현대고등학교 건너편  2020   6  20  14.066667   균형
21   강남구  현대고등학교 건너편  2020   6  21  13.400000   균형
22   강남구  현대고등학교 건너편  2020   6  22  15.000000   균형
23   강남구  현대고등학교 건너편  2020   6  23  17.133333   균형
24   강남구  현대고등학교 건너편  2020   7   0  10.433333  불균형
25   강남구  현대고등학교 건너편  2020   7   1  10.966667  불균형
26   강남구  현대고등학교 건너편  2020   7   2  11.133333   균형
27   강남구  현대고등학교 건너편  2020   7   3  11.566667   균형
28   강남구  현대고등학교 건너편  2020   7   4  12.400000   균형
29   강남구  현대고등학교 건너편  2020   7   5  12.200000   균형
30   강남구  현대고등학교 건너편  2020   7   6  12.300000   균형
31   강남구  현대고등학교 건너편  2020   7   7  12.666667   균형
32   강남구  현대고등학교 건너편  2020   7   8   7.366667  불균형
33   강남구  현대고등학교 건너편  2020   7   9   8.600000  불균형
34   강남구  현대고등학교 건너편  2020   7  10  10.066667  불균형
35   강남구  현대고등학교 건너편  2020   7  11  10.300000  불균형
36   강남구  현대고등학교 건너편  2020   7  12  11.333333   균형
37   강남구  현대고등학교 건너편  2020   7  13  11.266667   균형
38   강남구  현대고등학교 건너편  2020   7  14  11.333333   균형
39   강남구  현대고등학교 건너편  2020   7  15  11.100000   균형
40   강남구  현대고등학교 건너편  2020   7  16  10.500000  불균형
41   강남구  현대고등학교 건너편  2020   7  17   8.700000  불균형
42   강남구  현대고등학교 건너편  2020   7  18   8.033333  불균형
43   강남구  현대고등학교 건너편  2020   7  19   8.000000  불균형
44   강남구  현대고등학교 건너편  2020   7  20   8.400000  불균형
45   강남구  현대고등학교 건너편  2020   7  21   7.633333  불균형
46   강남구  현대고등학교 건너편  2020   7  22   9.533333  불균형
47   강남구  현대고등학교 건너편  2020   7  23  10.200000  불균형
48   강남구  현대고등학교 건너편  2020   8   0  13.033333   균형
49   강남구  현대고등학교 건너편  2020   8   1  13.433333   균형
50   강남구  현대고등학교 건너편  2020   8   2  13.966667   균형
51   강남구  현대고등학교 건너편  2020   8   3  14.333333   균형
52   강남구  현대고등학교 건너편  2020   8   4  14.566667   균형
53   강남구  현대고등학교 건너편  2020   8   5  14.600000   균형
54   강남구  현대고등학교 건너편  2020   8   6  14.700000   균형
55   강남구  현대고등학교 건너편  2020   8   7  14.866667   균형
56   강남구  현대고등학교 건너편  2020   8   8   7.600000  불균형
57   강남구  현대고등학교 건너편  2020   8   9   8.066667  불균형
58   강남구  현대고등학교 건너편  2020   8  10   9.166667  불균형
59   강남구  현대고등학교 건너편  2020   8  11   9.666667  불균형
60   강남구  현대고등학교 건너편  2020   8  12  10.266667  불균형
61   강남구  현대고등학교 건너편  2020   8  13  10.733333  불균형
62   강남구  현대고등학교 건너편  2020   8  14  11.133333   균형
63   강남구  현대고등학교 건너편  2020   8  15  11.433333   균형
64   강남구  현대고등학교 건너편  2020   8  16  11.333333   균형
65   강남구  현대고등학교 건너편  2020   8  17  10.366667  불균형
66   강남구  현대고등학교 건너편  2020   8  18  10.100000  불균형
67   강남구  현대고등학교 건너편  2020   8  19  10.133333  불균형
68   강남구  현대고등학교 건너편  2020   8  20  10.766667  불균형
69   강남구  현대고등학교 건너편  2020   8  21  10.833333  불균형
70   강남구  현대고등학교 건너편  2020   8  22  11.233333   균형
71   강남구  현대고등학교 건너편  2020   8  23  12.666667   균형
72   강남구  현대고등학교 건너편  2020   9   0  21.233333  불균형
73   강남구  현대고등학교 건너편  2020   9   1  22.100000  불균형
74   강남구  현대고등학교 건너편  2020   9   2  22.566667  불균형
75   강남구  현대고등학교 건너편  2020   9   3  22.933333  불균형
76   강남구  현대고등학교 건너편  2020   9   4  23.233333  불균형
77   강남구  현대고등학교 건너편  2020 

In [7]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구        대여소명     년   월  시간           p_0           p_1        L_s  \
0    강남구  현대고등학교 건너편  2020   6   0  3.669750e-02  4.023461e-02   7.955313   
1    강남구  현대고등학교 건너편  2020   6   1  1.363053e-01  1.213969e-01   4.695780   
2    강남구  현대고등학교 건너편  2020   6   2  1.255930e-01  1.137725e-01   4.943041   
3    강남구  현대고등학교 건너편  2020   6   3  8.096958e-03  1.048924e-02   9.999056   
4    강남구  현대고등학교 건너편  2020   6   4  2.110844e-03  3.107632e-03  10.944933   
5    강남구  현대고등학교 건너편  2020   6   5  5.629577e-02  5.830633e-02   7.067942   
6    강남구  현대고등학교 건너편  2020   6   6  3.740527e-02  4.091202e-02   7.919187   
7    강남구  현대고등학교 건너편  2020   6   7  8.314769e-02  8.116798e-02   6.109159   
8    강남구  현대고등학교 건너편  2020   6   8  5.265578e-03  7.115646e-03  10.363659   
9    강남구  현대고등학교 건너편  2020   6   9  2.013485e-04  3.642372e-04  11.767373   
10   강남구  현대고등학교 건너편  2020   6  10  9.472423e-04  1.498622e-03  11.304834   
11   강남구  현대고등학교 건너편  2020   6  11  2.148568e-02  2.510432e-02   8.848497   
12   강남구  현대고등학교 건너편  2020   6  12  2.220605e-02  2.584923e-02   8.799678   
13   강남구  현대고등학교 건너편  2020   6  13  2.501987e-03  3.626476e-03  10.852937   
14   강남구  현대고등학교 건너편  2020   6  14  7.482904e-02  7.429070e-02   6.382691   
15   강남구  현대고등학교 건너편  2020   6  15  1.283612e-01  1.157650e-01   4.877612   
16   강남구  현대고등학교 건너편  2020   6  16  1.787775e-01  1.494069e-01   3.852003   
17   강남구  현대고등학교 건너편  2020   6  17  7.753251e-02  7.654483e-02   6.291776   
18   강남구  현대고등학교 건너편  2020   6  18  1.097732e-01  1.020807e-01   5.339372   
19   강남구  현대고등학교 건너편  2020   6  19  1.183332e-01  1.084721e-01   5.120015   
20   강남구  현대고등학교 건너편  2020   6  20  3.198125e-02  3.566572e-02   8.206361   
21   강남구  현대고등학교 건너편  2020   6  21  9.702402e-02  9.226794e-02   5.689941   
22   강남구  현대고등학교 건너편  2020   6  22  2.433874e-02  2.803576e-02   8.659887   
23   강남구  현대고등학교 건너편  2020   6  23  8.901409e-03  1.142216e-02   9.908813   
24   강남구  현대고등학교 건너편  2020   7   0  6.823837e-02  6.871556e-02   6.613223   
25   강남구  현대고등학교 건너편  2020   7   1  3.411997e-02  3.774975e-02   8.090188   
26   강남구  현대고등학교 건너편  2020   7   2  1.375562e-01  1.222722e-01   4.667920   
27   강남구  현대고등학교 건너편  2020   7   3  8.098232e-02  7.939444e-02   6.178621   
28   강남구  현대고등학교 건너편  2020   7   4  1.584091e-04  2.924476e-04  11.820803   
29   강남구  현대고등학교 건너편  2020   7   5  5.294256e-01  2.491415e-01   0.888523   
30   강남구  현대고등학교 건너편  2020   7   6  3.020486e-01  2.114340e-01   2.237734   
31   강남구  현대고등학교 건너편  2020   7   7  4.927905e-03  6.701951e-03  10.413863   
32   강남구  현대고등학교 건너편  2020   7   8  1.170568e-03  1.817462e-03  11.220131   
33   강남구  현대고등학교 건너편  2020   7   9  4.568683e-04  7.704054e-04  11.552177   
34   강남구  현대고등학교 건너편  2020   7  10  1.141026e-04  2.165621e-04  11.888143   
35   강남구  현대고등학교 건너편  2020   7  11  3.779770e-02  4.128672e-02   7.899318   
36   강남구  현대고등학교 건너편  2020   7  12  6.551775e-03  8.667453e-03  10.187277   
37   강남구  현대고등학교 건너편  2020   7  13  9.054699e-02  8.714297e-02   5.880300   
38   강남구  현대고등학교 건너편  2020   7  14  6.516039e-02  6.607173e-02   6.725555   
39   강남구  현대고등학교 건너편  2020   7  15  1.117351e-01  1.035593e-01   5.288024   
40   강남구  현대고등학교 건너편  2020   7  16  1.084227e-01  1.010581e-01   5.375102   
41   강남구  현대고등학교 건너편  2020   7  17  1.774909e-01  1.486093e-01   3.874779   
42   강남구  현대고등학교 건너편  2020   7  18  1.020876e-01  9.620824e-02   5.547061   
43   강남구  현대고등학교 건너편  2020   7  19  7.266497e-02  7.247274e-02   6.456955   
44   강남구  현대고등학교 건너편  2020   7  20  5.837591e-02  6.014935e-02   6.984896   
45   강남구  현대고등학교 건너편  2020   7  21  1.086917e-01  1.012621e-01   5.367959   
46   강남구  현대고등학교 건너편  2020   7  22  1.918515e-02  2.270243e-02   9.010503   
47   강남구  현대고등학교 건너편  2020   7  23  3.748444e-02  4.098766e-02   7.915170   
48   강남구  현대고등학교 건너편  2020   8   0  2.510594e-03  3.637800e-03  10.851012   
49   강남구  현대고등학교 건너편  2020   8   1  3.625952e-02  3.981437e-02   7.977859   
50   강남구  현대고등학교 건너편  2020   8   2  5.265578e-03  7.115646e-03  10.363659   
5

In [8]:
len(df_new)

288

In [9]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구        대여소명     년   월  시간           p_0           p_1        L_s  \
0    강남구  현대고등학교 건너편  2020   6   0  3.669750e-02  4.023461e-02   7.955313   
1    강남구  현대고등학교 건너편  2020   6   1  1.363053e-01  1.213969e-01   4.695780   
2    강남구  현대고등학교 건너편  2020   6   2  1.255930e-01  1.137725e-01   4.943041   
3    강남구  현대고등학교 건너편  2020   6   3  8.096958e-03  1.048924e-02   9.999056   
4    강남구  현대고등학교 건너편  2020   6   4  2.110844e-03  3.107632e-03  10.944933   
5    강남구  현대고등학교 건너편  2020   6   5  5.629577e-02  5.830633e-02   7.067942   
6    강남구  현대고등학교 건너편  2020   6   6  3.740527e-02  4.091202e-02   7.919187   
7    강남구  현대고등학교 건너편  2020   6   7  8.314769e-02  8.116798e-02   6.109159   
8    강남구  현대고등학교 건너편  2020   6   8  5.265578e-03  7.115646e-03  10.363659   
9    강남구  현대고등학교 건너편  2020   6   9  2.013485e-04  3.642372e-04  11.767373   
10   강남구  현대고등학교 건너편  2020   6  10  9.472423e-04  1.498622e-03  11.304834   
11   강남구  현대고등학교 건너편  2020   6  11  2.148568e-02  2.510432e-02   8.848497   
12   강남구  현대고등학교 건너편  2020   6  12  2.220605e-02  2.584923e-02   8.799678   
13   강남구  현대고등학교 건너편  2020   6  13  2.501987e-03  3.626476e-03  10.852937   
14   강남구  현대고등학교 건너편  2020   6  14  7.482904e-02  7.429070e-02   6.382691   
15   강남구  현대고등학교 건너편  2020   6  15  1.283612e-01  1.157650e-01   4.877612   
16   강남구  현대고등학교 건너편  2020   6  16  1.787775e-01  1.494069e-01   3.852003   
17   강남구  현대고등학교 건너편  2020   6  17  7.753251e-02  7.654483e-02   6.291776   
18   강남구  현대고등학교 건너편  2020   6  18  1.097732e-01  1.020807e-01   5.339372   
19   강남구  현대고등학교 건너편  2020   6  19  1.183332e-01  1.084721e-01   5.120015   
20   강남구  현대고등학교 건너편  2020   6  20  3.198125e-02  3.566572e-02   8.206361   
21   강남구  현대고등학교 건너편  2020   6  21  9.702402e-02  9.226794e-02   5.689941   
22   강남구  현대고등학교 건너편  2020   6  22  2.433874e-02  2.803576e-02   8.659887   
23   강남구  현대고등학교 건너편  2020   6  23  8.901409e-03  1.142216e-02   9.908813   
24   강남구  현대고등학교 건너편  2020   7   0  6.823837e-02  6.871556e-02   6.613223   
25   강남구  현대고등학교 건너편  2020   7   1  3.411997e-02  3.774975e-02   8.090188   
26   강남구  현대고등학교 건너편  2020   7   2  1.375562e-01  1.222722e-01   4.667920   
27   강남구  현대고등학교 건너편  2020   7   3  8.098232e-02  7.939444e-02   6.178621   
28   강남구  현대고등학교 건너편  2020   7   4  1.584091e-04  2.924476e-04  11.820803   
29   강남구  현대고등학교 건너편  2020   7   5  5.294256e-01  2.491415e-01   0.888523   
30   강남구  현대고등학교 건너편  2020   7   6  3.020486e-01  2.114340e-01   2.237734   
31   강남구  현대고등학교 건너편  2020   7   7  4.927905e-03  6.701951e-03  10.413863   
35   강남구  현대고등학교 건너편  2020   7  11  3.779770e-02  4.128672e-02   7.899318   
36   강남구  현대고등학교 건너편  2020   7  12  6.551775e-03  8.667453e-03  10.187277   
37   강남구  현대고등학교 건너편  2020   7  13  9.054699e-02  8.714297e-02   5.880300   
38   강남구  현대고등학교 건너편  2020   7  14  6.516039e-02  6.607173e-02   6.725555   
39   강남구  현대고등학교 건너편  2020   7  15  1.117351e-01  1.035593e-01   5.288024   
40   강남구  현대고등학교 건너편  2020   7  16  1.084227e-01  1.010581e-01   5.375102   
41   강남구  현대고등학교 건너편  2020   7  17  1.774909e-01  1.486093e-01   3.874779   
42   강남구  현대고등학교 건너편  2020   7  18  1.020876e-01  9.620824e-02   5.547061   
43   강남구  현대고등학교 건너편  2020   7  19  7.266497e-02  7.247274e-02   6.456955   
44   강남구  현대고등학교 건너편  2020   7  20  5.837591e-02  6.014935e-02   6.984896   
45   강남구  현대고등학교 건너편  2020   7  21  1.086917e-01  1.012621e-01   5.367959   
46   강남구  현대고등학교 건너편  2020   7  22  1.918515e-02  2.270243e-02   9.010503   
47   강남구  현대고등학교 건너편  2020   7  23  3.748444e-02  4.098766e-02   7.915170   
48   강남구  현대고등학교 건너편  2020   8   0  2.510594e-03  3.637800e-03  10.851012   
49   강남구  현대고등학교 건너편  2020   8   1  3.625952e-02  3.981437e-02   7.977859   
50   강남구  현대고등학교 건너편  2020   8   2  5.265578e-03  7.115646e-03  10.363659   
51   강남구  현대고등학교 건너편  2020   8   3  3.485926e-02  3.846538e-02   8.050953   
52   강남구  현대고등학교 건너편  2020   8   4  3.574622e-02  3.932085e-02   8.004471   
53   강남구  현대고등학교 건너편  2020   8   5  4.706522e-01  2.491688e-01   1.123098   
5

In [10]:
len(z)

254

In [11]:
# TP

len(z[z['균형'] == '균형'])

143

In [12]:
# FN

len(z[z['균형'] == '불균형'])

111

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 111개

- 균형인데 균형이라고 판단한 수가 143개

In [13]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구        대여소명     년   월  시간           p_0           p_1        L_s  \
32   강남구  현대고등학교 건너편  2020   7   8  1.170568e-03  1.817462e-03  11.220131   
33   강남구  현대고등학교 건너편  2020   7   9  4.568683e-04  7.704054e-04  11.552177   
34   강남구  현대고등학교 건너편  2020   7  10  1.141026e-04  2.165621e-04  11.888143   
57   강남구  현대고등학교 건너편  2020   8   9  6.047062e-03  8.062750e-03  10.253977   
58   강남구  현대고등학교 건너편  2020   8  10  3.582116e-06  9.040579e-06  12.343783   
60   강남구  현대고등학교 건너편  2020   8  12  1.057950e-03  1.657456e-03  11.261432   
104  강남구  현대고등학교 건너편  2020  10   8  2.434165e-04  4.332814e-04  11.722318   
105  강남구  현대고등학교 건너편  2020  10   9  2.134965e-04  3.842937e-04  11.753736   
128  강남구  현대고등학교 건너편  2020  11   8  1.036688e-07  3.475953e-07  12.575001   
129  강남구  현대고등학교 건너편  2020  11   9  3.320082e-04  5.754808e-04  11.642702   
130  강남구  현대고등학교 건너편  2020  11  10  3.377790e-04  5.846175e-04  11.638050   
131  강남구  현대고등학교 건너편  2020  11  11  9.914291e-03  1.258352e-02   9.801257   
132  강남구  현대고등학교 건너편  2020  11  12  8.891277e-03  1.141047e-02   9.909922   
133  강남구  현대고등학교 건너편  2020  11  13  1.549571e-02  1.876945e-02   9.293515   
143  강남구  현대고등학교 건너편  2020  11  23  2.134965e-04  3.842937e-04  11.753736   
144  강남구  현대고등학교 건너편  2020  12   0  1.718631e-03  2.577946e-03  11.048122   
145  강남구  현대고등학교 건너편  2020  12   1  1.812537e-06  4.833433e-06  12.400015   
150  강남구  현대고등학교 건너편  2020  12   6  2.506645e-09  1.127990e-08  12.714286   
151  강남구  현대고등학교 건너편  2020  12   7  1.863473e-07  5.963112e-07  12.545456   
152  강남구  현대고등학교 건너편  2020  12   8  1.015330e-04  1.946049e-04  11.910642   
153  강남구  현대고등학교 건너편  2020  12   9  2.969617e-04  5.196831e-04  11.672210   
154  강남구  현대고등학교 건너편  2020  12  10  5.228375e-04  8.713958e-04  11.510980   
155  강남구  현대고등학교 건너편  2020  12  11  3.659415e-02  4.013552e-02   7.960620   
156  강남구  현대고등학교 건너편  2020  12  12  2.355464e-02  2.723505e-02   8.710496   
157  강남구  현대고등학교 건너편  2020  12  13  1.758453e-01  1.475845e-01   3.904142   
158  강남구  현대고등학교 건너편  2020  12  14  5.840627e-02  6.017616e-02   6.983697   
159  강남구  현대고등학교 건너편  2020  12  15  4.970589e-02  5.237825e-02   7.343414   
160  강남구  현대고등학교 건너편  2020  12  16  1.992398e-03  2.948750e-03  10.974778   
161  강남구  현대고등학교 건너편  2020  12  17  1.297139e-03  1.995599e-03  11.176583   
162  강남구  현대고등학교 건너편  2020  12  18  3.258201e-02  3.625322e-02   8.173315   
167  강남구  현대고등학교 건너편  2020  12  23  1.719999e-04  3.153332e-04  11.802890   
176  강남구  현대고등학교 건너편  2021   1   8  1.150093e-02  1.437617e-02   9.644052   
177  강남구  현대고등학교 건너편  2021   1   9  1.812537e-06  4.833433e-06  12.400015   
180  강남구  현대고등학교 건너편  2021   1  12  4.792628e-03  6.535402e-03  10.434516   

            W_s        재고량   균형  
32     8.868379   7.366667  불균형  
33     6.798504   8.600000  불균형  
34     7.279285  10.066667  불균형  
57    12.895450   8.066667  불균형  
58    17.634039   9.166667  불균형  
60    11.273358  10.266667  불균형  
104    7.035103   9.033333  불균형  
105    6.412498  10.833333  불균형  
128   22.191180   4.666667  불균형  
129   11.646569   5.500000  불균형  
130   13.433057   6.333333  불균형  
131    5.711194   6.900000  불균형  
132    4.999412   7.666667  불균형  
133    3.988644   8.233333  불균형  
143   14.107495  11.533333   균형  
144   41.501782   8.200000  불균형  
145  124.000377   8.466667  불균형  
150   95.357143   9.066667  불균형  
151   75.272748   9.433333  불균형  
152   29.779628   1.933333  불균형  
153   21.891895   2.466667  불균형  
154   23.034002   3.000000  불균형  
155    7.996449   3.400000  불균형  
156    8.363079   3.766667  불균형  
157    2.537757   3.733333  불균형  
158    3.371314   3.966667  불균형  
159    2.492747   4.366667  불균형  
160    4.398675   5.800000  불균형  
161    5.165123   7.133333  불균형  
162    3.569825   7.500000  불균형  
167   29.512300   8.100000  불균형  
176   73.171937   7.833333  불균형  
177   62.000189   8.233333  불균형  
180   14.297408   9.300000  불균형

In [14]:
len(w)

34

In [15]:
# FP

len(w[w['균형'] == '균형'])

1

In [16]:
# TN

len(w[w['균형'] == '불균형'])

33

- 불균형인데 균형이라고 판단한 수는 1
- 불균형인데 불균형이라고 판단한 수는 33

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [17]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.6111111111111112

In [18]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.9930555555555556

In [19]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.562992125984252